### ISSUE TITLE
FALSE NEGATIVE: `MC_OVERRIDABLE_METHOD_CALL_IN_CONSTRUCTOR` misses `Applet.getParameter()`

### ISSUE THREAD

**Issue by rikki322** [spotbugs/spotbugs#4308]
Hi, I found a false negative in SpotBugs 4.10.3 where `MC_OVERRIDABLE_METHOD_CALL_IN_CONSTRUCTOR` does not report a call to the overridable `Applet.getParameter()` method from a constructor.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs MC_OVERRIDABLE_METHOD_CALL_IN_CONSTRUCTOR`

**Minimal reproducer**

```java
import java.applet.Applet;

class SpotbugsSonarqubeAppletConstructor extends Applet {
  SpotbugsSonarqubeAppletConstructor() {
    getParameter("name");
  }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-sonarqube-mc-overridable-call-fn-jdk-method.java
spotbugs -textui -effort:max -low -visitors FindOverridableMethodCall classes
```

**Current behavior**

SpotBugs produces no `MC_OVERRIDABLE_METHOD_CALL_IN_CONSTRUCTOR` diagnostic for the call to `getParameter()`.

Running the full default analysis also produces no diagnostic for this program.

**Expected behavior**

SpotBugs should report the call at line 5.

`Applet.getParameter(String)` is a public, non-final instance method and can be overridden by subclasses. Calling it from a constructor therefore has the same risk as other overridable method calls during construction.

For comparison, `MC_OVERRIDABLE_METHOD_CALL_IN_CONSTRUCTOR` correctly reports other overridable methods called from equivalent constructors, including:

- inherited `Applet.getAppletInfo()`
- inherited `Applet.getCodeBase()`
- inherited `Component.getName()`
- ordinary user-defined overridable methods, including methods with arguments

Additionally, explicitly overriding `getParameter(String)` in the analyzed subclass does not make SpotBugs report the constructor call, suggesting that the false negative is specific to this method rather than inherited JDK methods in general.
